common/src/main/scala/org/apache/spark/sql/comet/util/Utils.scala

```diff
@@ -215,12 +215,8 @@ object Utils {
       val writer = new ArrowStreamWriter(root, provider, Channels.newChannel(out))
       writer.start()
       writer.writeBatch()
-
       root.clear()
-      writer.end()
-
-      out.flush()
-      out.close()
+      writer.close()
```

Comment:
Closing the writer will close the dictionary provider. If the dictionary arrays are shared across batches, you will close them and empty later batches. I remember we hit the issue before.

Reply:
Hmm... I need to take further look at this to fully understand if this fix is correct or not.

Reply:
This should be correct. The dictionary provider held by the writer [contains copied vectors](https://github.com/apache/arrow/blob/main/java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowStreamWriter.java#L159), so closing them does not interfere with the rest of the data structures.

Reply:
I remember closing it will cause some errors in CI due the reason I mentioned. Let's see if CI can pass or not.

Reply:
The CI passes on all commits of this PR, it has run 3 rounds without problem.

Reply:
Yea, maybe there was some other changes before. Anyway it is good to close the writer without issue.